### PR TITLE
[doc] Update doc with `cross-rs` and `musl` builds

### PR DESCRIPTION
**Pull request by jprendes**
This PR updates the mdbook documentation to take into account the changes introduced in #2536

### CODE REVIEW

**Comment by jprendes** [comment]
PTAL, thanks!
@utam0k @YJDoc2 @adrianalin

File: docs/src/user/basic_setup.md [hunk 1]
```diff
@@ -6,16 +6,35 @@ Youki currently only supports Linux Platform, and to use it on other platform yo
...
+```console
+$ just youki-dev # or youki-release
+```
 
-### Debian, Ubuntu and related distributions
+Install the build depdencies using your distribution's package manger
 
+#### Debian, Ubuntu and related distributions
 ```console
 $ sudo apt-get install    \
```
Comment (jprendes) [inline docs/src/user/basic_setup.md:39]: I'm not sure that we need all these dependencies.
IIUC we don't need neither of `libsystemd-dev`, `libelf-dev`, `libclang-dev` or `libssl-dev`.
Are they needed by any of the tests?

Comment (utam0k) [reply docs/src/user/basic_setup.md:39]: Good point. I often forget to update the dependencies. A good way to check is to delete them from there and attempt our CI. When all CI are passed, it means they aren't necessary.
https://github.com/containers/youki/blob/656ccc3f69329c2ed10129e046d0474e6c3dee13/justfile#L153-L160

**Review by utam0k — APPROVED** [review]
Good

**Pull request merged**